receiver/mongodbatlasreceiver/internal/metadata/metric_name_mapping.go

```diff
@@ -0,0 +1,1001 @@
+		"status": pdata.NewAttributeValueString("used"),
+		"aggregation": pdata.NewAttributeValueString("max"),
+	}},
+
+	// sfx: skipped
+	"DISK_PARTITION_SPACE_PERCENT_FREE": {"disk.partition.utilization", map[string]pdata.AttributeValue{
+		"status": pdata.NewAttributeValueString("free"),
+		"aggregation": pdata.NewAttributeValueString("avg"),
+	}},
+	// sfx: skipped
+	"MAX_DISK_PARTITION_SPACE_PERCENT_FREE": {"disk.partition.utilization", map[string]pdata.AttributeValue{
+		"status": pdata.NewAttributeValueString("free"),
+		"aggregation": pdata.NewAttributeValueString("max"),
+	}},
+	// sfx: skipped
+	"DISK_PARTITION_SPACE_PERCENT_USED": {"disk.partition.utilization", map[string]pdata.AttributeValue{
+		"status": pdata.NewAttributeValueString("used"),
+		"aggregation": pdata.NewAttributeValueString("avg"),
+	}},
+	// sfx: skipped
+	"MAX_DISK_PARTITION_SPACE_PERCENT_USED": {"disk.partition.utilization", map[string]pdata.AttributeValue{
+		"status": pdata.NewAttributeValueString("used"),
+		"aggregation": pdata.NewAttributeValueString("max"),
+	}},
+
+	// Process Database Measurements (https://docs.atlas.mongodb.com/reference/api/process-disks-measurements/)
+	// sfx: skipped
+	"DATABASE_COLLECTION_COUNT": {"db.counts", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("collection"),
+	}},
+	// sfx: skipped
+	"DATABASE_INDEX_COUNT": {"db.counts", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("index"),
+	}},
+	// sfx: skipped
+	"DATABASE_EXTENT_COUNT": {"db.counts", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("extent"),
+	}},
+	// sfx: skipped
+	"DATABASE_OBJECT_COUNT": {"db.counts", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("object"),
+	}},
+	// sfx: skipped
+	"DATABASE_VIEW_COUNT": {"db.counts", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("view"),
+	}},
+	// sfx: skipped
+	"DATABASE_AVERAGE_OBJECT_SIZE": {"db.size", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("object"),
+	}},
+	// sfx: skipped
+	"DATABASE_AVERAGE_STORAGE_SIZE": {"db.size", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("storage"),
+	}},
+	// sfx: skipped
+	"DATABASE_AVERAGE_INDEX_SIZE": {"db.size", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("index"),
+	}},
+	// sfx: skipped
+	"DATABASE_AVERAGE_DATA_SIZE": {"db.size", map[string]pdata.AttributeValue{
+		"type": pdata.NewAttributeValueString("data"),
+	}},
+}
+
+func mappedMetricByName(name string) (MetricIntf, map[string]pdata.AttributeValue) {
+	info, found := metricNameMapping[name]
+	if !found {
+		return nil, nil
+	}
+
+	metricinf := Metrics.ByName(info.metricName)
+	return metricinf, info.attributes
+}
+
+func inferMetricType(mongoType string) (pdata.MetricDataType, string) {
+	var dataType pdata.MetricDataType
+	switch mongoType {
+	case "BYTES", "MEGABYTES", "SECONDS", "KILOBYTES":
```

Comment:
Could these also be used to set the unit on the metric?

Reply:
They could. My initial intent was to use these values as a fallback for any not-currently-documented metric that gets added to Atlas from here on out, but I'm thinking that may be going too far and I should cut this functionality. I think that the metric format from Atlas differs enough from OTel standards that it might be counter-productive to pass them through without some more knowledge about them. I've also gotten some advice to keep this simple, and that functionality seems to be something that would be better added if needed later.